stac_fastapi/api/tests/test_app.py

```diff
@@ -0,0 +1,149 @@
+import importlib
+import os
+from datetime import datetime
+from typing import List, Optional, Union
+
+import pytest
+from fastapi.testclient import TestClient
+from pydantic import BaseModel
+
+from stac_fastapi.api.app import StacApi
+from stac_fastapi.api.models import create_get_request_model, create_post_request_model
+from stac_fastapi.extensions.core.filter.filter import FilterExtension
+from stac_fastapi.types import core, response_model, search
+from stac_fastapi.types.config import ApiSettings
+from stac_fastapi.types.search import BaseSearchPostRequest
+
+
+@pytest.fixture
+def cleanup():
+    old_environ = dict(os.environ)
+    yield
+    os.environ.clear()
+    os.environ.update(old_environ)
+
+
+@pytest.mark.parametrize(
+    "validate, response_type",
+    [
+        ("True", BaseModel),
+        ("False", dict),
+    ],
+)
+def test_app(validate, response_type, collection_dict, item_dict, cleanup):
+    os.environ["VALIDATE_RESPONSE"] = str(validate)
+    importlib.reload(response_model)
+    importlib.reload(core)
+
+    class MyCoreClient(core.BaseCoreClient):
+        def post_search(
+            self, search_request: BaseSearchPostRequest, **kwargs
+        ) -> response_model.ItemCollection:
+            return response_model.ItemCollection(
+                type="FeatureCollection", features=[response_model.Item(**item_dict)]
+            )
+
+        def get_search(
+            self,
+            collections: Optional[List[str]] = None,
+            ids: Optional[List[str]] = None,
+            bbox: Optional[List[Union[float, int]]] = None,
+            datetime: Optional[Union[str, datetime]] = None,
+            limit: Optional[int] = 10,
+            query: Optional[str] = None,
+            token: Optional[str] = None,
+            fields: Optional[List[str]] = None,
+            sortby: Optional[str] = None,
+            intersects: Optional[str] = None,
+            **kwargs,
+        ) -> response_model.ItemCollection:
+            # FIXME: hyphen alias for filter_crs and filter_lang are currently not working
+            # assert kwargs.get("filter_crs") == "EPSG:4326"
+            # assert kwargs.get("filter_lang") == "cql-test"
```

Comment:
I'm not sure to get this. Did you open an issue about it yet?

Reply:
yep, https://github.com/stac-utils/stac-fastapi/issues/638